plugins/kotlin/frontend-independent/tests/test/org/jetbrains/kotlin/test/utils/IgnoreTests.kt

```diff
@@ -126,15 +126,7 @@ object IgnoreTests {
             true -> "passes"
         }
 
-        val directiveIsOutdated = when {
-            !testPasses && directive is EnableOrDisableTestDirective.Enable -> true
-            testPasses && directive is EnableOrDisableTestDirective.Disable -> true
-            else -> false
-        }
-
-        val directiveIsMissing = !directiveIsOutdated
-
-        if (directiveIsMissing && INSERT_DIRECTIVE_AUTOMATICALLY) {
+        if (INSERT_DIRECTIVE_AUTOMATICALLY && directive is EnableOrDisableTestDirective.Enable) {
```

Comment:
The fix seems to be incorrect as it now handles only the case when the directive is `// FIR_COMPARISON` is used to mark the test as passing, but some tests uses `// FIR_IGNORE` to mark tests as failing, in that case `directive is EnableOrDisableTestDirective.Disable`

Reply:
This logic is only trying to automatically fix the test if it passes. This aligns with the previous code, which doesn't offer to fix the test by removing `// IGNORE_FIR`.

I can add the ability to remove `// IGNORE_FIR` if that's desirable though.

Reply:
Thanks let me update it.